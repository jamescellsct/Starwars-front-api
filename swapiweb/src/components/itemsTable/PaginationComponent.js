import React, { Component } from 'react';
import {
    Pagination,
    PaginationItem,
    PaginationLink
} from 'reactstrap';

class PaginationComponent extends Component {

    handlePagination(data) {
        let {onSelected} = this.props;

        let arr = data.page > 0 && data.page < 3 ? [1,2,3] :
            data.page > 2 && data.page < data.pageCount-1 ? [data.page-1, data.page, data.page+1] : 
                [data.pageCount-2, data.pageCount-1, data.pageCount];

        return arr.map((item, i) =>
            <PaginationItem key={i} active={item === data.page}>
                <PaginationLink onClick={() => onSelected(item, `${data.object}`.charAt(0).toLowerCase() + `${data.object}`.slice(1))}>
                    {item}
                </PaginationLink>
            </PaginationItem>
        )
    }

    render() {
        let data = this.props.data;
        let { onSelected } = this.props;
        return (
            <Pagination>
                <PaginationItem {...data.page === 1 ? 'disabled' : ''}>
                    <PaginationLink first='true' onClick={() => onSelected(1, `${data.object}`.charAt(0).toLowerCase() + `${data.object}`.slice(1))} >
                        First
                    </PaginationLink>
                </PaginationItem>
                {
                    this.handlePagination(data)
                }
                <PaginationItem {...data.page === data.pageCount ? 'disabled' : ''}>
                    <PaginationLink last='true' onClick={() => onSelected(data.pageCount, `${data.object}`.charAt(0).toLowerCase() + `${data.object}`.slice(1))} >
                        Last
                    </PaginationLink>
                </PaginationItem>
            </Pagination>
        );
    }
}

export default PaginationComponent;