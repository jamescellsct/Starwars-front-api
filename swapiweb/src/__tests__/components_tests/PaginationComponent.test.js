import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import PaginationComponent from "../../components/itemsTable/PaginationComponent";

Enzyme.configure({ adapter: new Adapter() });

describe("Testing PaginationComponent", () => {
  it("Should render correctly without state", () => {
    const wrapper = Enzyme.shallow(<PaginationComponent />);

    expect(wrapper).toMatchSnapshot();
  });
});
