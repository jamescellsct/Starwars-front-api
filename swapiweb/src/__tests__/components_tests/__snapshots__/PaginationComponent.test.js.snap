// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PaginationComponent with state Should render correctly with redux state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Connect(PaginationComponent))
      resource="peoples"
    />
  </Router>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "resource": "peoples",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "resource": "peoples",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "router": Object {
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "route": Object {
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "match": Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        },
      },
    },
  },
}
`;

exports[`Testing PaginationComponent with state Should render correctly without redux state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(PaginationComponent)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
